### PR TITLE
Configurable nix options and substitute-on-destination

### PR DESCRIPTION
**Pull request by johanot**
Trying to make morph more configurable by allowing for pass-through of nix.conf options. It's now possible to provide an attribute set of configuration options (`nixConfig`) for each deployment. These options will be passed to both `nix-build` and `nix copy`. Unfortunately, because of the way morph does things, it's not possible to set nix config on a per-host level.

In an effort to also try and fix #73, i've added `substituteOnDestination` as a deployment module option. This option applies with per-host granularity.

While working on this, I considered just making the entire nix command-line available to the user, which would make any possible flag and option configurable. I rejected that idea for now though, because it turns out to be complicated. There are many permutations of per-host and per-deployment options and there should be a way in the API to then also distinguish flags intended for `nix-build` and `nix copy` or even `nix eval`. 

TLDR; I figured that would make the API too complicated at this point. I hope that this PR at least solves "some" problems.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -151,6 +151,33 @@ Health checks will be repeated until success, and the interval can be configured
 
 It is currently possible to have expressions like `"test \"$(systemctl list-units --failed --no-legend --no-pager |wc -l)\" -eq 0"` (count number of failed systemd units, fail if non-zero) as the first argument in a cmd-healthcheck. This works, but is discouraged, and might break at any time.
 
+### Advanced configuration
+
+**nix.conf-options:** The "network"-attrset supports a sub-attrset named "nixConfig". Options configured here will pass `--option <name> <value>` to all nix commands.
+Note: these options apply to an entire deployment and are *not* configurable on per-host basis.
+The default is an empty set, meaning that the nix configuration is inherited from the build environment. See `man nix.conf`.
```
[the remark this answers is not in the document]
Comment (johanot) [reply README.md]: It only overrides stuff in the context of the morph build itself. It never overrides environment config permanently. Nothing is leaking.

[the remark this answers is not in the document]
Comment (adamtulinius) [reply README.md]: But it can override options set outside of morph that might now just be gone? I don't really think it's an issue, just wondering.

[the remark this answers is not in the document]
Comment (johanot) [reply README.md]: I'm not sure I get what you're trying to say. It sets `--option foo bar` on the cmdline for nix-build. So yeah, that takes precedence over anything set in the local `nix.conf` or whatever. Which... Is sort-of the point with this feature :)

[the remark this answers is not in the document]
Comment (adamtulinius) [reply README.md]: At any rate it's documented now, so maybe I should just leave it at that :)

**Comment by johanot** [comment]
Sure! Let's see if we can get #91 squeezed in the release as well. :) I'll get back to you on that.

**Comment by flokli** [comment]
Thanks!
